ensime_shared/protocol.py

```diff
@@ -232,10 +232,10 @@ def handle_doc_uri(self, call_id, payload):
 
     def handle_completion_info_list(self, call_id, payload):
         """Handler for a completion response."""
+        self.log.debug('handle_completion_info_list: in')
         completions = payload["completions"]
-        self.log("handle_completion_info_list: in")
         self.suggestions = [completion_to_suggest(c) for c in completions]
-        self.log("handle_completion_info_list: {}".format(self.suggestions))
+        self.log.debug('handle_completion_info_list: %s', self.suggestions)
```

Comment:
Yeah, I'm aware of the cost of the string generation. A possible solution could be to wrap data-like log args with  something like this:

```python
class DataFmt(object):
    def __init__(self, data):
        self.data = data

    def __str__(self):
        return pprint.pformat(self.data)

self.log.debug("blah %s", new DataFMT(dictionary))
```

Remember, I'm not a python dev, so it might seem a little too JVMish :)